### PR TITLE
Open Paying for College data APIs to access from any origin

### PR DESCRIPTION
**Pull request by Scotchester**
This PR removes CORS restrictions on the Paying for College data APIs so that they can be accessed from any origin. This will be necessary for the forthcoming standalone version of Your Financial Path to Graduation (a.k.a. "the Grad Path tool").

I've tested this locally by way of my in-development standalone front end running on localhost:3000, accessing APIs running on a local cf.gov environment running on localhost:8000. Prior to this change, the front end could not access the APIs, because the different port is enough for it to be considered a different origin.

---

## Additions

- Signal handler to bypass CORS on requests beginning with `/paying-for-college2/understanding-your-financial-aid-offer/api/`

## How to test this PR

I have opened [the first PR to github.com/cfpb/grad-path](https://github.com/cfpb/grad-path/pull/1) with a functioning standalone front end prototype. Using that, you will be able to test this PR.

## Notes

Based on this example in the django-cors-headers docs: https://github.com/adamchainz/django-cors-headers#signals

## Checklist

- [x] PR has an informative and human-readable title
  - PR titles are used to generate the change log in [releases](../../releases); good ones make that easier to scan.
  - Consider prefixing, e.g., "Mega Menu: fix layout bug", or "Docs: Update Docker installation instructions".
- [x] Changes are limited to a single goal (no scope creep)
- [x] Code follows the standards laid out in the [CFPB development guidelines](https://github.com/cfpb/development)
- [ ] Future todos are captured in comments and/or tickets
- [ ] Project documentation has been updated, potentially one or more of:
  - [This repo’s docs](https://cfpb.github.io/consumerfinance.gov/) (edit the files in the `/docs` folder) – for basic, close-to-the-code docs on working with this repo
  - CFGOV/platform wiki on GHE – for internal CFPB developer guidance
  - CFPB/hubcap wiki on GHE – for internal CFPB design and content guidance

### CODE REVIEW

File: cfgov/paying_for_college/handlers.py [hunk 1]
```diff
@@ -0,0 +1,6 @@
+from corsheaders.signals import check_request_enabled
+
+def cors_allow_particular_urls(sender, request, **kwargs):
```
Comment (willbarton) [inline cfgov/paying_for_college/handlers.py:3]: One terribly minor thing, could we include a comment above this handler about why we're allowing any origin here (to explicitly permit the use of this API from outside), and maybe a note that that's what this is doing, allowing all origins (the `check_request_enabled` signal name isn't exactly descriptive of that, and this is the first time we're using a pattern like this).

**Review by willbarton — APPROVED** [review]
One really minor thing, requesting some comments, and then I'll get this merged.

**Comment by anselmbradford** [comment]
@Scotchester if you fix the backend linting issue I can merge this in.

**Comment by Scotchester** [comment]
> @Scotchester if you fix the backend linting issue I can merge this in.

Ah dangit, stupid quotes.

**Comment by Scotchester** [comment]
OK, I guess it wasn't just that. I'll have to get Black to actually run in my environment this weekend.

**Comment by anselmbradford** [comment]
> OK, I guess it wasn't just that. I'll have to get Black to actually run in my environment this weekend.

Ah too bad. @willbarton do you see what the linter issue is?

**Comment by higs4281** [comment]
Black adds two blank lines:
- one above `def cors_allow_particular_urls(sender, request, **kwargs):`
- one above `check_request_enabled.connect(cors_allow_particular_urls):`

**Comment by Scotchester** [comment]
Thanks, Bill! Tried that last night, but the checks seem stalled :\

**Comment by Scotchester** [comment]
Gracias, @anselmbradford!

**Pull request merged**